Dockerfile

```diff
@@ -1,15 +1,22 @@
 FROM registry.redhat.io/ubi8/ubi-minimal
 
-WORKDIR /app_root/src
+RUN microdnf install --disableplugin=subscription-manager --nodocs -y python38 tar gzip
 
-COPY Pipfile Pipfile.lock manage.py seed.py gunicorn.conf.py ./
+# Install poetry in separate virtual env
+ENV POETRY_HOME=/opt/poetry
+RUN python -m venv $POETRY_HOME &&\
+    $POETRY_HOME/bin/pip install poetry
 
-RUN microdnf install --disableplugin=subscription-manager --nodocs -y python38 tar gzip
+# Set new virtual env for ROS
+ENV VIRTUAL_ENV=/app_root/src \
+    PATH="/app_root/src/bin:$POETRY_HOME/bin:$PATH"
 
+WORKDIR /app_root/src
+COPY pyproject.toml poetry.lock manage.py seed.py gunicorn.conf.py ./
 COPY ros ros
 COPY migrations migrations
 COPY seed.d seed.d
 
-RUN pip3 install --upgrade pip setuptools wheel && \
-    pip3 install pipenv && \
-    pipenv install --system --deploy --ignore-pipfile
+RUN python -m venv $VIRTUAL_ENV && \
```

Comment:
@upadhyeammit Do we need a venv for installs in container, cause `poetry install` creates one?

Reply:
Let me try that! but if I remember correctly then it did not work for me if I don't create and change to virtenv before running poetry cmds!

Reply:
as you can see the services does not come up if i remove the virt env,

https://ci.int.devshift.net/job/RedHatInsights-ros-backend-pr-check/606/console

if you have any suggestions on doing it without creating the virt env upfront then let me know